library-extension-rx/src/main/java/com/instacart/library/truetime/TrueTimeRx.java

```diff
@@ -180,7 +201,23 @@ public void accept(long[] ntpResponse) {
                           public Flowable<InetAddress> apply(String ntpPoolAddress) {
                               try {
                                   TrueLog.d(TAG, "---- resolving ntpHost : " + ntpPoolAddress);
-                                  return Flowable.fromArray(InetAddress.getAllByName(ntpPoolAddress));
+                                  InetAddress[] addresses = InetAddress.getAllByName(ntpPoolAddress);
+                                  Predicate<InetAddress> filter = _addressFilter;
+                                  if (filter == null) {
+                                      return Flowable.fromArray(addresses);
+                                  } else {
+                                      ArrayList<InetAddress> validAddresses = new ArrayList<>();
+                                      for (InetAddress i : addresses) {
+                                          try {
+                                              if (filter.test(i)) {
+                                                  validAddresses.add(i);
+                                              }
+                                          } catch (Exception e) {
+                                              return Flowable.error(e);
+                                          }
+                                      }
+                                      return Flowable.fromIterable(validAddresses);
```

Comment:
Just wondering - what's the background behind using `Flowable.fromIterable` here and `Flowable.fromArray` above?

Reply:
addresses are stored in an array, whereas validAddresses are represented by a List (Iterable), so I believe that's what makes the difference